force-app/main/default/classes/HouseholdNamingService.cls

```diff
@@ -196,6 +186,24 @@ public without sharing class HouseholdNamingService {
         }
     }
 
+    private Boolean qualifiesForHouseholdRenaming(SObject household, HouseholdName householdName, String formalGreeting, String informalGreeting) {
+        return !household.get('Name').equals(householdName.asName())
+                || formalGreeting != null
+                && !formalGreeting.equals(String.valueOf(
+                householdName.asFormalGreeting()))
+                || informalGreeting != null
+                && !informalGreeting.equals(String.valueOf(
+                householdName.asInformalGreeting()));
+    }
+
+    private List<Contact> getHouseholdMembers(Map<Id, List<Contact>> membersByHouseholdId, SObject household) {
+        List<Contact> householdMembers =
+                membersByHouseholdId.get(idFor(household)) == null ?
+                        new List<Contact>() :
+                        membersByHouseholdId.get(idFor(household));
```

Comment:
This logic can be expensive because it issues two `get()`'s against the map. How about:
```
List<Contact> householdMembers = membersByHouseholdId.get(idFor(household));
if (householdMembers == null) {
    householdMembers = new List<Contact>();
}
```